From 16e7db892ef8aa5d28e6457481b9d77708069653 Mon Sep 17 00:00:00 2001
From: Madhu <enometh@net.meer>
Date: Fri, 22 Sep 2017 17:23:17 +0530
Subject: [PATCH 5/6] xterm-decomposed_bitmaps.patch

Source: opensuse
 From: Egbert Eich <eich@suse.de>
 Date: Tue Jun 3 18:17:07 2014 +0200
 Subject: [PATCH]Fix number of colormap entries for DirectColor and TrueColor visuals
 Patch-mainline: to be upstreamed
 Git-commit: 071525466531009223f1c21a40aa2e081d5ef99a
 References: bnc#

DirectColor and TrueColor have a decomposed colormap,
therefore the colormap_size field in the XVisualInfo
structure is meant per sub-field (take a look at the output
of xdpyinfo).
With 3 sub-fields there are colormap_size^3 entires.

Signed-off-by: Egbert Eich <eich@suse.de>

---
 misc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/misc.c b/misc.c
index cbb2679..f00c1de 100644
--- a/misc.c
+++ b/misc.c
@@ -2692,6 +2692,8 @@ getColormapInfo(XtermWidget xw, unsigned *typep, unsigned *sizep)
     if (getVisualInfo(xw)) {
 	*typep = (unsigned) xw->visInfo->class;
 	*sizep = (unsigned) xw->visInfo->colormap_size;
+	if (*typep > 3)
+		*sizep = (*sizep)*(*sizep)*(*sizep);
     } else {
 	*typep = 0;
 	*sizep = 0;
-- 
2.35.1.dirty

